packages/compass-components/src/components/cancel-loader.spec.tsx

```diff
@@ -23,6 +24,10 @@ describe('ConfirmationModal Component', function () {
     spy = sinon.spy();
   });
 
+  afterEach(function () {
+    cleanup();
```

Comment:
Ahh, interesting, clean up should be set up by default, but seems like [this is indeed needed when running tests in watch mode](https://testing-library.com/docs/react-testing-library/setup#auto-cleanup-in-mochas-watch-mode). I think it might be worth having in our shared config so we don't need to manually add it everywhere, what do you think?

Reply:
Hmm I didn't think of looking for a global solution. That's much better. I'll change it.

Reply:
OK. I have tried that but I cannot get it to work. I added console.log() calls and I can see that hook file loading and I can even see that afterEach() executing, but it is definitely not cleaning up.

I thought that it was requiring a different copy of @testing-library/react and therefore cleanup() cleaned up already empty state (and not the state used by the tests) but I then made sure that we use the exact same patch version of testing library everywhere (in package.json in package-lock.json, etc) and I even did a git clean -fdx && npm run bootstrap but it is still not working.

I do a console.log(require.resolve('@testing-library/react')); in configs/mocha-config-compass/register/mocha-watch-cleanup-after-each.js and it prints
/Users/leroux.bodenstein/mongo/compass/configs/mocha-config-compass/node_modules/@testing-library/react/dist/index.js

That's probably a clue as to what's going on, but I don't know how to get around it.

So I'm going to just file a ticket and leave this alone for now. It sucks a bit, but at least it works.

Reply:
See https://jira.mongodb.org/browse/COMPASS-5240?filter=-2